Runtime/ObjectPool.cs

```diff
@@ -0,0 +1,130 @@
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+// ReSharper disable InconsistentNaming
+
+namespace Yonii8.Unity.ObjectPooling
+{
+    
+    [CreateAssetMenu(fileName = "Unnamed Pool", menuName = "Yonii/Object Pooling/Create New Pool")]
+    public class ObjectPool : ScriptableObject
+    {
+        private List<GameObject> _objects;
+        private GameObject _parent;
+        private bool _initialised;
+        
+        [SerializeField] private int _initialCount;
+        
+        public GameObject Prefab;
+        public bool ExpandablePool;
+
+        public void Initialise(Transform poolManager)
+        {            
+            _parent = new GameObject(name: $"{Prefab.name}_Pool");
+            _parent.transform.SetParent(poolManager, worldPositionStays: false);
+
+            Prefab.SetActive(false);
+
+            if (_initialCount == 0)
+            {
+                Debug.LogWarning(
+                    "Initial count has been found as 0. It will be defaulted to 5, please check that you have properly populated the value.");
+                _initialCount = 5;
+            }
+            _objects = FillPool(_initialCount);
+            Prefab.SetActive(true);
+        }
+
+        public GameObject GetPooledObject()
+        {
+            if (!_initialised)
+            {
+                Debug.LogWarning(
+                    "Pool has not initialised yet!" +
+                    "Please make sure that whatever objects you are grabbing are done post initialising!"
+                    );
+
+                throw new ApplicationException();
```

Comment:
Return null as I do not want the game to flop due to pool not being initialised.